src/containers/wallet/ConnectionStatus.tsx

```diff
@@ -0,0 +1,121 @@
+import greenColor from '@material-ui/core/colors/green';
+import redColor from '@material-ui/core/colors/red';
+import yellowColor from '@material-ui/core/colors/yellow';
+import {
+  createStyles,
+  Theme,
+  withStyles,
+  WithStyles
+} from '@material-ui/core/styles';
+import * as classNames from 'classnames';
+import { inject, observer } from 'mobx-react';
+import * as React from 'react';
+import { defineMessages, InjectedIntlProps, injectIntl } from 'react-intl';
+import { LoadingState } from '../../stores/account';
+import WalletStore from '../../stores/wallet';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    connectionIcon: {
+      width: '1em',
+      height: '1em',
+      borderRadius: '50%',
+      marginRight: '0.5em'
+    },
+    connected: {
+      backgroundColor: greenColor[800]
+    },
+    disconnected: {
+      backgroundColor: redColor[800]
+    },
+    connecting: {
+      backgroundColor: yellowColor[800]
+    },
+    text: {
+      [theme.breakpoints.down('xs')]: {
+        display: 'none'
+      }
+    }
+  });
+
+interface Props extends WithStyles<typeof styles> {}
+
+interface PropsInjected extends Props {
+  walletStore: WalletStore;
+}
+
+type DecoratedProps = Props & InjectedIntlProps;
+
+const stylesDecorator = withStyles(styles, { name: 'ConnectionStatus' });
+
+const messages = defineMessages({
+  connectedStatus: {
+    id: 'wallet-connection.connected-status',
+    description: 'Connection status when connected',
+    defaultMessage: 'Connected'
+  },
+  disconnectedStatus: {
+    id: 'wallet-connection.disconnected-status',
+    description: 'Connection status when disconnected',
+    defaultMessage: 'Disconnected'
+  },
+  connectingStatus: {
+    id: 'wallet-connection.connecting-status',
+    description: 'Connection status when connecting',
+    defaultMessage: 'Connecting'
+  }
+});
+
+@inject('walletStore')
+@observer
+class ConnectionStatus extends React.Component<DecoratedProps> {
+  get injected(): PropsInjected & DecoratedProps {
+    // @ts-ignore
+    return this.props;
+  }
+
+  render() {
+    const { intl, classes, walletStore } = this.injected;
+
+    switch (walletStore.connected) {
+      case LoadingState.LOADED:
+        return (
+          <React.Fragment>
+            <span
+              className={classNames(classes.connected, classes.connectionIcon)}
+            />
+            <span className={classes.text}>
```

Comment:
Incorrect font. Should either have it as `<Typography>` element, or include the theme font settings in to the `text` style.

Reply:
Why isnt the correct font set globally? Fixed here and opened #196.

Reply:
@TobiaszCudnik I've just gone with how material-ui expects things to be done. I suspect there's a reason why they do it explicitly (I would imagine some earlier version set the font globally and at some point they changed it to explicit tag), but I don't recall actually looking into what it might be.